musit-models/src/main/scala/no/uio/musit/models/Module.scala

```diff
@@ -0,0 +1,25 @@
+package no.uio.musit.models
+
+import play.api.libs.json.{JsNumber, Reads, Writes, __}
+
+sealed trait Module {
+  val id: Int
+}
+
+case object StorageFacility extends Module {
+  val id = 1
+}
+case object LoanProcess extends Module {
+  val id = 2
+}
+
+object Module {
+
+  implicit val reads: Reads[Module]   = __.read[Int].map(fromInt)
+  implicit val writes: Writes[Module] = Writes{m => JsNumber(m.id)}
+
+  def fromInt(id: Int): Module = id match {
+    case StorageFacility.id => StorageFacility
```

Comment:
This function is incomplete in the match since it doesn't account for arguments that doesn't match any of the predefined Modules. Return type should then also be `Option[Module]`

If you need a function that should return a non Option Module, you still need to account for ID's that doesn't match.
The function is then `unsafe` and should be defined as follows:

```scala
@throws(classOf[NoSuchElementException])
def unsafeFromInt(id: Int): Module = fromInt(id).get
```
For a similar example, see `MusitUUIDOps`.